### PR TITLE
Added dark mode support to ProfileEditActivity, ProfileListActivity, ProfileRenameActivity, ProfileResetPinActivity and AppVersionActivity

### LINKED ISSUE #4176: Add dark mode support to ProfileEditActivity, ProfileListActivity, ProfileRenameActivity, ProfileResetPinActivity and AppVersionActivity.
Add dark mode support to ProfileEditActivity, ProfileListActivity, ProfileRenameActivity, ProfileResetPinActivity and AppVersionActivity.

Go through this [wiki](https://github.com/oppia/oppia-android/wiki/Dark-Mode) to get started.

#### Location in the app
1. Open the app
2. Go to Administrator Controls
3. Go to edit profile
4. These activities can be found navigated to through there.

### PR DESCRIPTION
**Pull request by aayushimathur6**
## Explanation
 Fixes #4176

Profile List : https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/ddc3dadf-10e4-45f6-aeea-eb8a330e00ff/specs/

Profile Edit : https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/693301de-5907-4f6a-b183-037412478267/

Profile Rename : https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/d0227921-1c24-4bb7-8667-a8fa5f4742dd/

Profile Reset : https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/32c9276d-072c-4edf-abf9-435323bb475a/

App Version : https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/79e0846d-de5c-42b7-93aa-f93d4995616d/

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

| BEFORE | AFTER |
| ------------- | ------------- |
| image | image  |
| image | image  |
| image |image  |
| image |image |
| image | image |

### CODE REVIEW

File: app/src/main/res/drawable/general_item_background_border_grey.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+  android:shape="rectangle">
+  <solid android:color="@color/component_color_profile_rename_activity_secondary_background_color" />
+  <stroke
+    android:width="1dp"
+    android:color="@color/color_def_black_12" />
+</shape>
```
Comment (Sarthak2601) [inline app/src/main/res/drawable/general_item_background_border_grey.xml:9]: EOF missing.

**Review by Sarthak2601 — COMMENTED** [review]
Thanks @aayushimathur6! I took a pass on this PR and had one nit suggestion. 
Apart from that I also suggest going through the "Essential Checklist" above and making that sure you do all of those tasks. I can see two of those items have not been fulfilled in this PR -- proper PR naming and assigning.

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 1 above)
Comment (rt4914) [inline app/src/main/res/drawable/general_item_background_border_grey.xml:9]: EOF is missing, add one blank line at the end.

File: app/src/main/res/drawable/general_item_background_border.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <shape xmlns:android="http://schemas.android.com/apk/res/android"
   android:shape="rectangle">
-  <solid android:color="@color/color_def_white" />
+  <solid android:color="@color/component_color_profile_list_activity_profile_view_background_color" />
```
Comment (rt4914) [inline app/src/main/res/drawable/general_item_background_border.xml]: Suggest recreating this color independent of activity ... something like
`component_color_general_item_background_border_solid_color`

File: app/src/main/res/drawable/selector_app_version.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
     <layer-list>
       <item android:left="1dp" android:top="3dp">
         <shape>
-          <solid android:color="@color/color_def_white" />
+          <solid android:color="@color/component_color_app_version_activity_container_background_color" />
```
Comment (rt4914) [inline app/src/main/res/drawable/selector_app_version.xml]: Rename to `component_color_selector_app_version_solid_color `

File: app/src/main/res/drawable/start_button_transparent_background.xml [hunk 4]
```diff
@@ -6,6 +6,9 @@
         <shape>
           <corners android:radius="4dp" />
           <solid android:color="@color/oppia_disabled_button_background" />
+          <stroke
+            android:width="1dp"
+            android:color="@color/component_color_profile_rename_activity_inactive_save_button_border_color" />
```
Comment (rt4914) [inline app/src/main/res/drawable/start_button_transparent_background.xml]: Rename to `component_color_start_button_transparent_background_sroke_color `

File: app/src/main/res/layout-land/profile_edit_fragment.xml [hunk 5]
```diff
@@ -54,6 +54,7 @@
         android:layout_marginStart="24dp"
         android:importantForAccessibility="no"
         android:text="@{viewModel.profile.name}"
+        android:textColor="@color/component_color_profile_edit_activity_profile_name_text_viewer_color"
```
Comment (rt4914) [inline app/src/main/res/layout-land/profile_edit_fragment.xml]: Rename to `component_color_profile_edit_activity_profile_edit_name_text_view_color `

File: app/src/main/res/layout-land/profile_edit_fragment.xml [hunk 6]
```diff
@@ -85,15 +86,15 @@
         style="?android:attr/borderlessButtonStyle"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="@color/color_def_white"
+        android:background="@color/component_color_profile_edit_activity_profile_rename_button_color"
```
Comment (rt4914) [inline app/src/main/res/layout-land/profile_edit_fragment.xml]: Rename to `component_color_profile_edit_activity_profile_rename_button_background_color `

File: app/src/main/res/layout-land/profile_edit_fragment.xml [hunk 7]
```diff
@@ -113,15 +114,15 @@
         style="?android:attr/borderlessButtonStyle"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="@color/color_def_white"
+        android:background="@color/component_color_profile_edit_activity_profile_reset_pin_button_color"
```
Comment (rt4914) [inline app/src/main/res/layout-land/profile_edit_fragment.xml]: Rename to `component_color_profile_edit_activity_profile_reset_pin_button_background_color `

File: app/src/main/res/layout-land/profile_edit_fragment.xml [hunk 8]
```diff
@@ -140,7 +141,7 @@
         android:id="@+id/profile_edit_allow_download_container"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="@color/color_def_white"
+        android:background="@color/component_color_profile_edit_activity_profile_edit_allow_download_container"
```
Comment (rt4914) [inline app/src/main/res/layout-land/profile_edit_fragment.xml:144]: Rename to `component_color_profile_edit_activity_profile_edit_allow_download_container_background_color`

File: app/src/main/res/layout/profile_edit_fragment.xml [hunk 9]
```diff
@@ -53,6 +53,7 @@
         android:layout_height="wrap_content"
         android:layout_marginStart="24dp"
         android:layout_marginEnd="16dp"
+        android:textColor="@color/component_color_profile_edit_activity_profile_name_text_viewer_color"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_edit_fragment.xml]: Rename to `component_color_profile_edit_activity_profile_edit_name_text_view_color `

File: app/src/main/res/layout/profile_edit_fragment.xml [hunk 10]
```diff
@@ -87,15 +88,15 @@
         style="?android:attr/borderlessButtonStyle"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="@color/color_def_white"
+        android:background="@color/component_color_profile_edit_activity_profile_rename_button_color"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_edit_fragment.xml]: Ditto as earlier comment for another file.

File: app/src/main/res/layout/profile_edit_fragment.xml [hunk 11]
```diff
@@ -87,15 +88,15 @@
...
         android:fontFamily="sans-serif"
         android:gravity="start|center_vertical"
         android:minHeight="48dp"
         android:paddingStart="16dp"
         android:paddingEnd="16dp"
         android:text="@string/profile_edit_rename"
         android:textAllCaps="false"
-        android:textColor="@color/oppia_primary_text"
+        android:textColor="@color/component_color_profile_edit_activity_profile_rename_button_text_color"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_edit_fragment.xml:99]: Ditto as earlier comment for another file.

File: app/src/main/res/layout/profile_edit_fragment.xml [hunk 12]
```diff
@@ -115,15 +116,15 @@
         style="?android:attr/borderlessButtonStyle"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="@color/color_def_white"
+        android:background="@color/component_color_profile_edit_activity_profile_reset_pin_button_color"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_edit_fragment.xml]: Ditto as earlier comment for another file.

File: app/src/main/res/layout/profile_edit_fragment.xml [hunk 13]
```diff
@@ -142,7 +143,7 @@
         android:id="@+id/profile_edit_allow_download_container"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="@color/color_def_white"
+        android:background="@color/component_color_profile_edit_activity_profile_edit_allow_download_container"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_edit_fragment.xml:146]: Ditto as earlier comment for another file.

File: app/src/main/res/layout/profile_list_profile_view.xml [hunk 14]
```diff
@@ -41,7 +41,7 @@
       android:layout_marginEnd="20dp"
       android:fontFamily="sans-serif"
       android:text="@{profile.name}"
-      android:textColor="@color/oppia_primary_text_dark"
+      android:textColor="@color/component_color_profile_list_activity_profile_name_color"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_list_profile_view.xml]: `component_color_profile_list_activity_profile_name_text_color `

File: app/src/main/res/layout/profile_list_profile_view.xml [hunk 15]
```diff
@@ -57,7 +57,7 @@
       android:layout_marginEnd="20dp"
       android:fontFamily="sans-serif"
       android:text="@string/profile_chooser_admin"
-      android:textColor="@color/color_def_accessible_light_grey"
+      android:textColor="@color/color_palette_profile_view_designation_color"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_list_profile_view.xml]: We should only use colors from `component_colors.xml` file.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@aayushimathur6 Suggested changes.

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 1 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/general_item_background_border_grey.xml:9]: done @Sarthak2601

Comment (aayushimathur6) [reply app/src/main/res/drawable/general_item_background_border_grey.xml:9]: done

File: app/src/main/res/drawable/general_item_background_border.xml (hunk 2 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/general_item_background_border.xml]: done

File: app/src/main/res/drawable/selector_app_version.xml (hunk 3 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/selector_app_version.xml]: updated

File: app/src/main/res/drawable/start_button_transparent_background.xml (hunk 4 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/start_button_transparent_background.xml]: updated

Comment (aayushimathur6) [reply app/src/main/res/drawable/start_button_transparent_background.xml]: done

File: app/src/main/res/layout-land/profile_edit_fragment.xml (hunk 5 above)
Comment (aayushimathur6) [reply app/src/main/res/layout-land/profile_edit_fragment.xml]: done

File: app/src/main/res/layout-land/profile_edit_fragment.xml (hunk 6 above)
Comment (aayushimathur6) [reply app/src/main/res/layout-land/profile_edit_fragment.xml]: done

File: app/src/main/res/layout-land/profile_edit_fragment.xml (hunk 7 above)
Comment (aayushimathur6) [reply app/src/main/res/layout-land/profile_edit_fragment.xml]: done

File: app/src/main/res/layout-land/profile_edit_fragment.xml (hunk 8 above)
Comment (aayushimathur6) [reply app/src/main/res/layout-land/profile_edit_fragment.xml:144]: done

File: app/src/main/res/layout/profile_edit_fragment.xml (hunk 9 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/profile_edit_fragment.xml]: updated

File: app/src/main/res/layout/profile_edit_fragment.xml (hunk 10 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/profile_edit_fragment.xml]: updated

File: app/src/main/res/layout/profile_edit_fragment.xml (hunk 11 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/profile_edit_fragment.xml:99]: updated

File: app/src/main/res/layout/profile_edit_fragment.xml (hunk 12 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/profile_edit_fragment.xml]: done

File: app/src/main/res/layout/profile_edit_fragment.xml (hunk 13 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/profile_edit_fragment.xml:146]: updated

File: app/src/main/res/layout/profile_list_profile_view.xml (hunk 14 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/profile_list_profile_view.xml]: updated

File: app/src/main/res/layout/profile_list_profile_view.xml (hunk 15 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/profile_list_profile_view.xml]: done

**Comment by aayushimathur6** [comment]
@rt4914 updated this PR with all the suggested changes, Sorry for the delay

**Comment by aayushimathur6** [comment]
@rt4914, @Sarthak2601 committed all the suggested changes, sorry for delay PTAL thanks

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/drawable/general_item_background_border_grey.xml:9]: Still pending

File: app/src/main/res/values-night/color_palette.xml [hunk 16]
```diff
@@ -1,13 +1,17 @@
...
 <resources>
   <color name="color_palette_primary_color">@color/color_def_oppia_turquoise</color>
+  <color name="color_palette_primary_toolbar_color">@color/color_def_forest_green</color>
   <color name="color_palette_primary_dark_color">@color/color_def_oppia_bangladesh_green</color>
   <color name="color_palette_accent_color">@color/color_def_oppia_metallic_blue</color>
   <color name="color_palette_primary_text_color">@color/color_def_oppia_silver</color>
+  <color name="color_palette_secondary_text_color">@color/color_def_oppia_silver</color>
   <color name="color_palette_highlighted_text_color">@color/color_def_white</color>
   <color name="color_palette_background_color">@color/color_def_oppia_light_black</color>
+  <color name="color_palette_primary_background_color">@color/color_def_oppia_light_black</color>
```
Comment (rt4914) [inline app/src/main/res/values-night/color_palette.xml]: This color is already defined in this file. So pelase remove this one

**Review by rt4914 — CHANGES_REQUESTED** [review]
@aayushimathur6 PTAL at all open comment threads.

File: app/src/main/res/values-night/color_palette.xml (hunk 16 above)
Comment (aayushimathur6) [reply app/src/main/res/values-night/color_palette.xml]: done

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 1 above)
Comment (Sarthak2601) [reply app/src/main/res/drawable/general_item_background_border_grey.xml:9]: still pending. 
add a blank line at the end.

**Comment by aayushimathur6** [comment]
updated

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 1 above)
Comment (Sarthak2601) [reply app/src/main/res/drawable/general_item_background_border_grey.xml:9]: It's still pending but I think @rt4914 has also commented on the same line, so I'll resolve this conversation for now.

File: app/src/main/res/drawable/general_item_background_border_black.xml [hunk 17]
```diff
@@ -5,4 +5,5 @@
   <stroke
     android:width="1dp"
     android:color="@color/component_color_general_item_background_stroke_color" />
+
```
Comment (rt4914) [inline app/src/main/res/drawable/general_item_background_border_black.xml]: Undo this change.

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/drawable/general_item_background_border_grey.xml:9]: Still pending.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Merge with latest develop and see why CI checks are failing @aayushimathur6

File: app/src/main/res/drawable/general_item_background_border_black.xml (hunk 17 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/general_item_background_border_black.xml]: done

**Review by rt4914 — CHANGES_REQUESTED** [review]
@aayushimathur6 There are definitely some issues with CI checks.

Please follow these steps for now:
1. Merge with latest develop.
2. Run and check the app again
3. Create screenshots showing `error-text` for input-box for the screen on which you have worked.
4. Once you have done all these, please ping me so that I can have a look at fresh CI failure s if any.

**Comment by aayushimathur6** [comment]
Closing this pull request as the project in build code was messed up and therefore making a new PR for it. Thanks
